### PR TITLE
[OSDEV-1492] Moderation. Moderation Queue Page (integration)- it is possible to send invalid date through UI

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-1492](https://opensupplyhub.atlassian.net/browse/OSDEV-1492) Moderation. Moderation Queue Page (integration)- it is possible to send invalid date through UI

- Invalid dates are now trimmed and properly handled
- Fixed formatting timezone issue with date

[OSDEV-1492]: https://opensupplyhub.atlassian.net/browse/OSDEV-1492?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by roman-stolar** [comment]
@coderabbit review

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 1]
```diff
@@ -65,6 +75,8 @@ const DashboardModerationQueue = ({
     const wasNotEmptyAndNowEmpty = (prev, current) =>
         prev?.current?.length > 0 && current?.length === 0;
 
+    const isValidDate = date => !date || dateRegexFormat.test(date);
```
Comment (VadimKovalenkoSNF) [inline src/react/src/components/Dashboard/DashboardModerationQueue.jsx:78]: I suggest to use this validation:
`const isValidDate = date => typeof date === 'string' && dateRegexFormat.test(date);`
It ensures that input is string and rejects `null`, `undefined`, `numbers`, etc.

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 2]
```diff
@@ -112,41 +124,57 @@ const DashboardModerationQueue = ({
...
-            );
-            fetchEvents();
-        } else {
+        if (!isValidDate(date)) {
+            dispatch(updateBeforeDate(''));
+            setErrorDateText(DATE_FORMAT_ERROR);
+            setBeforeDateError(true);
+            return;
+        }
+        if (date && afterDate && date < afterDate) {
```
Comment (VadimKovalenkoSNF) [inline src/react/src/components/Dashboard/DashboardModerationQueue.jsx:160]: I've noticed double usage of this check. What about to move it into a separate function?

**Review by VadimKovalenkoSNF — CHANGES_REQUESTED** [review]
Few minor suggestions.

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx (hunk 2 above)
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx]: Done.

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx (hunk 1 above)
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx]: Done.

Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx]: Also a litle bit reworked due to requirements

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx (hunk 2 above)
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx]: Also a litle bit reworked due to requirements

File: src/react/src/components/Dashboard/DashboardModerationQueue.jsx [hunk 3]
```diff
@@ -50,13 +57,16 @@ const DashboardModerationQueue = ({
...
+    const [errorDateText, setErrorDateText] = useState('');
     const isFirstRender = useRef(true);
     const prevDataSources = useRef(dataSources);
     const prevModerationStatuses = useRef(moderationStatuses);
     const prevCountries = useRef(countries);
 
     const dispatch = useDispatch();
 
+    const dateRegexFormat = /^\d{4}-\d{2}-\d{2}$/;
+
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/components/Dashboard/DashboardModerationQueue.jsx:69]: The logic is right.

File: src/react/src/util/util.js [hunk 4]
```diff
@@ -1314,7 +1314,7 @@ export function sort(array, comparator) {
     return stabilizedThis.map(el => el[0]);
 }
 
-export const formatDate = (date, format) => moment(date).format(format);
+export const formatDate = (date, format) => moment.utc(date).format(format);
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/react/src/util/util.js:1317]: I checked there is different scenarios. I didn't break anythings that use those formatDate function.

**Review by VadimKovalenkoSNF — APPROVED** [review]
LGTM

**Review by vladsha-dev — APPROVED** [review]
LGTM

**Pull request merged**